src/gui/plugins/scene3d/Scene3D.cc

```diff
@@ -2811,7 +2812,57 @@ void Scene3D::OnDropped(const QString &_drop, int _mouseX, int _mouseY)
   math::Vector3d pos = renderWindow->ScreenToScene({_mouseX, _mouseY});
 
   msgs::EntityFactory req;
-  req.set_sdf_filename(_drop.toStdString());
+  std::string dropStr = _drop.toStdString();
+  if (QUrl(_drop).isLocalFile())
+  {
+    // mesh to sdf model
+    common::rtrim(dropStr);
+    std::string lowerStr = common::lowercase(dropStr);
+
+    if (!common::EndsWith(lowerStr, ".dae")
```

Comment:
Rather than implementing this logic here, use `MeshManager::IsValidFilename`: https://github.com/ignitionrobotics/ign-common/blob/ign-common4/graphics/src/MeshManager.cc#L192

Reply:
bf7731282f9e13485fbf6f6a360d0997fac861fb